plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,496 @@
+        dnac_host: "{{dnac_host}}"
+        dnac_username: "{{dnac_username}}"
+        dnac_password: "{{dnac_password}}"
+        dnac_verify: "{{dnac_verify}}"
+        dnac_port: "{{dnac_port}}"
+        dnac_version: "{{dnac_version}}"
+        dnac_debug: "{{dnac_debug}}"
+        dnac_log: True
+        dnac_log_level: DEBUG
+        config_verify: True
+        dnac_api_task_timeout: 1000
+        dnac_task_poll_interval: 1
+        accesspoints: "{{ accesspoints }}"
+      register: output_list
+    - name: iterate through module output (a list)
+      debug:
+        msg: '{{ item }}'   
+        with_items: "{{output_list.output }}"
+"""
+
+RETURN = r"""
+#Case: Modification of the AP details updated and Rebooted Accesspoint
+response:
+  description: A dictionary with activation details as returned by the Catalyst Center Python SDK
+  returned: always
+  type: dict
+  sample: >
+    {
+        "response": {
+            "taskId": "string",
+            "url": "string"
+        },
+        "version": "string"
+    }
+"""
+
+class DnacAutomation(DnacBase):
+    """Class containing member attributes for DNAC Access Point Automation module"""
+
+    def __init__(self, module):
+        super().__init__(module)
+        self.result["response"] = []
+        self.supported_states = ["merged"]
+        self.payload = module.params
+        self.baseurl = "https://" + module.params["dnac_host"]+ ":" + module.params["dnac_port"]
+        self.log('Login DNAC using by user: ' + module.params["dnac_username"], "INFO")
+        try:
+            self.dnac = api.DNACenterAPI(base_url=self.baseurl, 
+                                    username = module.params["dnac_username"],
+                                    password = module.params["dnac_password"],
+                                    verify = False)
+        except Exception as e:
+            self.log("Unable to Login DNAC "+ str(e) , "ERROR")
+
+    # Below function used pydentic validation over the ansible validation
+    # We can customize validation, if it not required we can remove this function.
+    def validate_input(self, inputdata):
+        """
+        Validate the fields provided in the yml files.
+        Checks the configuration provided in the playbook against a predefined specification
+        to ensure it adheres to the expected structure and data types based pydentics package.
+        Parameters:
+          - inputdata: To validate the input file of yaml keys values will be validated.
+        Returns:
+          The method not returns anything just validation input if anything worng will stop execution.
+        Example:
+            To use this method, create an instance of the class and call 'validate_input' on it.
+          If the validation succeeds, this will allow to go next step, unless this will stop execution.
+          based on the fields.
+        """
+        self.log('Validating the Yaml File..', "INFO")
```

Comment:
For validation of the input we create a dictionary called {name of the workflow manager}_spec and put all the inputs and their type.

Like in this case it will be 
accepoint_spec = {"management_ip_address": {"type":str, "required": True},
                                "macAddress": {"type":str, "required": True}}

valid_accesspoint, invalid_params = validate_list_of_dicts(
            self.config, accepoint_spec
        )


if invalid_params:
            self.msg = "Invalid parameters in playbook: {0}".format(
                "\n".join(invalid_params))
            self.log(str(self.msg), "ERROR")
            self.status = "failed"
            return self

Reply:
This is handled in the pydentic validation. try catch block will show the param error over the Ansible validation.